### PR TITLE
Add long error explanation for E0728

### PR DESCRIPTION
**Pull request by JohnTitor**
Part of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc/error_codes.rs [hunk 1]
```diff
@@ -2126,6 +2126,74 @@ static X: u32 = 42;
...
+
+`await` is used to suspend the current computation until the given
+future is ready to produce a value. So it is legal only within
+an async context, like an `async fn` or an `async` block.
+
+```edition2018
+# use std::pin::Pin;
+# use std::future::Future;
+# use std::task::{Context, Poll};
+
```
Comment (estebank) [inline src/librustc/error_codes.rs]: Won't these empty lines render as empty code lines in the docs?

File: src/librustc/error_codes.rs [hunk 2]
```diff
@@ -2126,6 +2126,74 @@ static X: u32 = 42;
...
+#             self.0 = true;
+#             Poll::Pending
+#         }
+#     }
+# }
+
+async fn foo() {
+    wake_and_yield_once().await // `await` is used within `async` context
+}
+```
```
Comment (estebank) [inline src/librustc/error_codes.rs]: Add an example inside of an `async` block, I'd say.

File: src/librustc/error_codes.rs (hunk 1 above)
Comment (JohnTitor) [reply src/librustc/error_codes.rs]: Indeed, added `#`s

File: src/librustc/error_codes.rs (hunk 2 above)
Comment (JohnTitor) [reply src/librustc/error_codes.rs]: Added an `async` block example, what about?

Comment (estebank) [reply src/librustc/error_codes.rs]: Great.

My last nitpick would be to have a link to the async/await documentation, but this is already a great improvement.

Comment (JohnTitor) [reply src/librustc/error_codes.rs]: Hm, at glance, we don't have the links to keywords in error code's explanations. I found the links to RFCs or specific topics. We should have a link to the async/await?

Comment (estebank) [reply src/librustc/error_codes.rs]: I would lean towards over linking over under linking, but I'll defer to @GuillaumeGomez

**Review by GuillaumeGomez — APPROVED** [review]

File: src/librustc/error_codes.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/librustc/error_codes.rs]: I'd prefer to have the links indeed.

**Comment by GuillaumeGomez** [comment]
Once the links have been added, just r=me

File: src/librustc/error_codes.rs (hunk 2 above)
Comment (JohnTitor) [reply src/librustc/error_codes.rs]: Okay, the link should be https://doc.rust-lang.org/std/keyword.async.html and https://doc.rust-lang.org/std/keyword.await.html ?

**Pull request merged**